mtags/src/main/scala-3/scala/meta/internal/pc/completions/MatchCaseCompletions.scala

```diff
@@ -241,38 +237,64 @@ object CaseKeywordCompletion:
       )
       completionOption.foreach(result += _)
     }
-
-    val basicMatch = CompletionValue.MatchCompletion(
-      "match",
-      Some(
-        if clientSupportsSnippets then "match\n\tcase$0\n"
-        else "match"
-      ),
-      Nil,
-      "",
-    )
     val members = result.result()
-    val completions = members match
-      case Nil => List(basicMatch)
-      case head :: tail =>
-        val insertText = Some(
-          tail
-            .map(_.label)
-            .mkString(
-              if clientSupportsSnippets then s"match\n\t${head.label} $$0\n\t"
-              else s"match\n\t${head.label}\n\t",
-              "\n\t",
-              "\n",
+    if !isFromPartial then
```

Comment:
code above can be simplified to
```scala
    val sortedSubclasses = subclassesForType(tpe)
    val members = sortedSubclasses.flatMap { case sym =>
      val autoImports = autoImportsGen.forSymbol(sym).getOrElse(Nil)
      completionGenerator.toCompletionValue(
        sym = sym,
        name = sym.decodedName,
        autoImports = autoImports,
      )
    }
```
there is no need to listBuffer

Reply:
You're right, but @dos65 is doing more changes on this part in https://github.com/scalameta/metals/pull/4397, so I would leave it here as it is